### PR TITLE
Removed the overwrite flag for syncing files

### PR DESCRIPTION
**Pull request by etorreborre**
It is too dangerous to leave the possibility to overwrite an existing file when synchronizing files to a given location.

### CODE REVIEW

File: notion-distcopy/src/main/scala/com/ambiata/notion/distcopy/SynchronizedInputsOutputs.scala [hunk 1]
```diff
@@ -8,6 +8,7 @@ import com.ambiata.notion.core._
 import com.ambiata.saws.s3.S3Pattern
 import org.apache.hadoop.conf.Configuration
 import org.apache.hadoop.fs._
+import scala.reflect.macros.whitebox
```
Comment (charleso) [inline notion-distcopy/src/main/scala/com/ambiata/notion/distcopy/SynchronizedInputsOutputs.scala]: ?

Comment (etorreborre) [inline notion-distcopy/src/main/scala/com/ambiata/notion/distcopy/SynchronizedInputsOutputs.scala]: Automatic imports, yay!

File: notion-distcopy/src/main/scala/com/ambiata/notion/distcopy/SynchronizedInputsOutputs.scala [hunk 2]
```diff
@@ -149,16 +150,16 @@ object SynchronizedInputsOutputs {
     }
 
   /** @return the list of all copied files */
-  def copyFromS3ToHdfs(source: S3Location, target: HdfsLocation, overwrite: Boolean, locationIO: LocationIO): RIO[List[Location]] =
+  def copyFromS3ToHdfs(source: S3Location, target: HdfsLocation, locationIO: LocationIO): RIO[List[Location]] =
     locationIO.isDirectory(source) >>= { isDirectory =>
-      if (!isDirectory) locationIO.copyFile(source, target, overwrite).as(List(source))
+      if (!isDirectory) locationIO.exists(target).flatMap(exist => locationIO.syncFile(source, target).whenM(!exist)).as(List(source))
```
Comment (charleso) [inline notion-distcopy/src/main/scala/com/ambiata/notion/distcopy/SynchronizedInputsOutputs.scala]: Is this duplicated - doesn't `syncFile` do the same check?

Comment (etorreborre) [inline notion-distcopy/src/main/scala/com/ambiata/notion/distcopy/SynchronizedInputsOutputs.scala]: Yes, I did this refactoring in 2 steps, but forgot to clean up that code, thanks.

**Comment by markhibberd** [comment]
@etorreborre Quick question on this, how does this behaviour bubble up and affect clients, on the surface (I could be wrong) it looks like:
 - The old behaviour was - overwrite or fail.
 - The new behaviour is - ignore clashes.

For a general usefulness perspective, I think there are three operations:
  1. Fail if the file already exists (this is generally the default for most things).
  2. Soft drop, and ignore when  target exists - in this case we probably want some way of communicating that the ignore has happened - inline with the "fail" or "monitor" philosophy I guess.
  3. Overwrite.

I am very ok with saying (3) is fairly dangerous and is pretty much never what we want to do - which is what this change does, so :+1:  for that. But it just makes me wonder if can we still do (1) or (2), and if so, what is the default (or are they totally separate calls), and does this change affect any clients who thought they were getting (1) or (3) but will now get (2) -- this would be bad, the rename helps considerably here, but just want to be sure we are covering all the downstream affects when we make these sort of changes.

**Comment by etorreborre** [comment]
What we can do (and should) do is to log which files are not being synchronized because they are already there. A bit like what gets displayed with distcopy mappings. 

And thinking about it I think we should discriminate: 
 
 - files being downloaded to the sync dir -> log
 - files being uploaded to a result dir -> fail

Agree?

**Comment by markhibberd** [comment]
@etorreborre Yes. With the extra note that the "log" should be something that we can read/monitor/take a copy of for historical purposes, so probably needs to be more that just printing to terminal. With that I think things are a lot more robust so :100:

**Comment by etorreborre** [comment]
@markhibberd in 23d9b360 I've added some hard fail + console printing corresponding to our discussion above.

**Pull request merged**